compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -491,6 +491,9 @@ impl Instruction {
             }
             Instruction::Truncate { value, bit_size, max_bit_size } => {
                 if let Some((numeric_constant, typ)) = dfg.get_numeric_constant_with_type(*value) {
+                    if bit_size == max_bit_size {
+                        return SimplifiedTo(*value);
+                    }
```

Comment:
```suggestion
            Instruction::Truncate { value, bit_size, max_bit_size } => {
                if bit_size == max_bit_size {
                    return SimplifiedTo(*value);
                 }
                if let Some((numeric_constant, typ)) = dfg.get_numeric_constant_with_type(*value) {
```

This applies to all truncations surely?

Reply:
done